pageserver/src/basebackup.rs

```diff
@@ -126,25 +126,22 @@ fn parse_rel_file_path(path: &str) -> Result<(), FilePathError> {
         Ok(())
     } else if let Some(dbpath) = path.strip_prefix("base/") {
         let mut s = dbpath.split('/');
-        let dbnode_str = s
-            .next()
-            .ok_or_else(|| FilePathError::new("invalid relation data file name"))?;
+        let dbnode_str = s.next().ok_or(FilePathError::InvalidFileName)?;
         let _dbnode = u32::from_str_radix(dbnode_str, 10)?;
```

Comment:
`let _dbnode: u32 = dbnode_str.parse()` (clippy complains)

Reply:
My clippy is happy. But yeah, I can see that check in the clippy sources (https://github.com/rust-lang/rust-clippy/blob/master/clippy_lints/src/from_str_radix_10.rs). Maybe I need to update.

I changed these to use parse with the turbofish notation (`str.parse::<u32>()`), even though the compiler could infer the type in these cases. I feel it's too subtle to leave it out in this case.